src/calculation.py

```diff
@@ -3,13 +3,13 @@
 from numpy import *
 
 
-def create_dataset(x_start, x_stop, equation, step_size):
+def create_dataset(x_start: float, x_stop: float, equation, step_size: float):
```

Comment:
Good with the type hinting, eventually we'd probably want to move to type hinting overall, as well as to give an expected return type. That can be done in a seperate PR, but I don't mind adding these types along thew way as we touch the code.